public_html/wp-content/mu-plugins/blocks/source/blocks/schedule/schedule-grid.js

```diff
@@ -0,0 +1,331 @@
+	} );
+}
+
+/**
+ * Derive the tracks so that they're easily accessible.
+ *
+ * The relevant terms are buried in the REST API response and not easily accessible, so this is just a convenience
+ * function so that they're easy to use.
+ *
+ * @param {Array} sessions
+ *
+ * @return {Array}
+ */
+function deriveSessionTracks( sessions ) {
+	// This can't run twice because it deletes the data that would be needed for the subsequent runs.
+	if ( sessions[0].derived && sessions[0].derived.tracks ) {
+		return sessions;
+	}
+
+	return sessions.map( session => {
+		const taxonomies = session._embedded[ 'wp:term' ] || [];
+		const tracks     = taxonomies.filter( terms => terms.length && 'wcb_track' === terms[ 0 ].taxonomy );
+
+		session.derived.tracks = tracks[0];
+			// todo why is it nested an extra layer? any side-effects of ignoring that?
+
+		// Remove raw tracks so that the derived data is canonical and DRY.
+		delete session._embedded[ 'wp:term' ];
+		// todo will deleting things here affect other blocks if the data is reused? if so, then maybe this block should make a copy in memory at some point higher up the chain
+			// primary motivation to reuse is to avoid an unnecessary HTTP request, rather than to save on a small amount of memory
+			// same applies to deleting entries in deriveSessionStartEndTimes()
+
+		return session;
+	} );
+}
+
+/**
+ * Get the tracks that the given Sessions are assigned to.
+ *
+ * In general, it's more useful for this block to know which tracks are assigned, rather than which ones exist,
+ * because e.g., you don't want to print all tracks in `GridColumnHeaders`, only the ones being used.
+ *
+ * @param {Array} sessions
+ *
+ * @return {Array}
+ */
+function getPopulatedTracks( sessions ) {
+	const tracks = sessions.reduce( ( tracks, session ) => {
+		if ( ! session.derived.tracks ) {
+			console.log(session);
+			// todo some session.derived.tracks are undefined
+		}
+
+		// todo some start/end times are invalid too - not anymore now that solved 'return early' bug?
+
+		for ( const track of session.derived.tracks ) {
+			tracks[ track.id ] = track;
+		}
+
+		return tracks;
+	}, [] );
+
+	// todo this results in tracks.length being "47", or whatever the highest `track.id` is plus 1
+		// using tracks.push() would result in duplicated being added, though.
+		// is it "wrong" to use an assoc array instead of sequentially numerically indexed?
+		// sorting below switches to sequentially numbered, but leaves length at 47
+
+	// todo is this necessary b/c that's what print_dynamic_styles does?
+		// or can it be removed?
+		// if removing, that returns back to when array was indexed by id instead of sequentially, is that a problem?
+	tracks.sort( ( a, b ) => {
+		return a.slug > b.slug;
+	} );
+
+	return tracks;
+}
+
+// todo
+export class ScheduleGrid extends Component {
+	// does this need to be a class component? only if it has state? if so, are there other blocks which don't need to be Components?
```

Comment:
That comment was more shorthand for "are we going to need any state or lifecycle methods in the future? if not, then come back at the end and make this a functional component".

I could go ahead and make it functional now, and then switch back to stateful in the future if there's a need; there's nothing wrong with that approach, the only reason I haven't done is is because I've been focused on the more important work of getting the basic features built, and didn't want to get distracted by tangential minor issues.